Composer/packages/client/src/pages/design/Listener.tsx

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React, { useEffect } from 'react';
+import { globalHistory, RouteComponentProps } from '@reach/router';
+import { PromptTab } from '@bfc/shared';
+import { useRecoilValue } from 'recoil';
+
+import { getDialogData } from '../../utils/dialogUtil';
+import { getFocusPath } from '../../utils/navigation';
+import { dispatcherState, currentDialogState } from '../../recoilModel';
+import { decodeDesignerPathToArrayPath } from '../../utils/convertUtils/designerPathEncoder';
+
+const getTabFromFragment = () => {
+  const tab = window.location.hash.substring(1);
+
+  return Object.values(PromptTab).find((value) => tab === value);
+};
+
+const Listener: React.FC<RouteComponentProps<{ dialogId: string; projectId: string; skillId?: string }>> = (props) => {
```

Comment:
Please pick a better name for this or at least add comments describing the responsibilities of this faceless component.

Even better, consider creating a hook.